services/src/main/java/org/keycloak/credential/WebAuthnCredentialProviderFactory.java

```diff
@@ -16,23 +16,39 @@
 
 package org.keycloak.credential;
 
+import com.webauthn4j.converter.util.ObjectConverter;
+import com.webauthn4j.data.client.Origin;
 import org.keycloak.Config;
 import org.keycloak.common.Profile;
 import org.keycloak.models.KeycloakSession;
-
-import com.webauthn4j.converter.util.ObjectConverter;
-import org.keycloak.models.KeycloakSessionFactory;
 import org.keycloak.provider.EnvironmentDependentProviderFactory;
 
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
 public class WebAuthnCredentialProviderFactory implements CredentialProviderFactory<WebAuthnCredentialProvider>, EnvironmentDependentProviderFactory {
 
     public static final String PROVIDER_ID = "keycloak-webauthn";
 
     private ObjectConverter converter;
+    private Config.Scope config;
+
+    @Override
+    public void init(Config.Scope config) {
+        this.config = config;
+    }
 
     @Override
     public CredentialProvider create(KeycloakSession session) {
-        return new WebAuthnCredentialProvider(session, createOrGetObjectConverter());
+        Set<Origin> origins = Optional.ofNullable(config.getArray("fido2-origins"))
```

Comment:
In the admin console, there is tab `Webauthn Policy` (and also `WebAuthn Passwordless policy`) under tabs `Authentication` and subtab `Policies`. It will be nice to have this configurable at those tabs instead of configuring this at provider level. As also all the other things related to WebAuthn are configured there at the policy level. Is it possible to update this PR according to it?

Reply:
Do we want to wait first to see issues around the same issue? I mean, having it as a provider option makes it possible to support native-apps without necessarily exposing it through our UIs.

Not sure. That is what I had in mind.